### ISSUE TITLE
Support for label and references

### ISSUE THREAD

**Issue by ketozhang** [mitya57/python-markdown-math#19]
This might not be the correct place to ask this however python markdown math fails to render `\label` and `\ref`.

A simple example given by [Mathjax](http://docs.mathjax.org/en/latest/tex.html) is,

```latex
In equation \eqref{eq:sample}, we find the value of an
interesting integral:

\begin{equation}
  \int_0^\infty \frac{x^3}{e^x-1}\,dx = \frac{\pi^4}{15}
  \label{eq:sample}
\end{equation}
````
Interestingly `\tag` works perfectly but it's not so useful to me when I want to reference.

As a motivation my idea of markdown math is to be able to render everything seen in the [Feynman Lectures](http://www.feynmanlectures.caltech.edu/III_02.html). The use of MathJax here simple but amazing which some features are missing in every Markdown math extension I've used so far.

**Comment by mitya57** [comment]
It works fine here, but there are two gotchas:

* Equation numbers must be explicitly enabled in the MathJax config as described in the [documentation](http://docs.mathjax.org/en/latest/options/input-processors/TeX.html). For example this is a config that works:

    ```html
    <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX", "output/HTML-CSS", "output/NativeMML"],
      extensions: ["MathMenu.js", "MathZoom.js"],
      TeX: {
        equationNumbers: {autoNumber: "AMS"}
      }
    });
    </script>
    ```
* In the first line, you need to add the delimiters around `\eqref{eq:sample}`, like this: `\(\eqref{eq:sample}\)`.

After you do these two steps, it will work:
![screenshot](https://user-images.githubusercontent.com/1381584/34460593-ff254a04-ee23-11e7-97e6-b4eefba840e8.png)
